klog.go

```diff
@@ -953,6 +961,7 @@ func (l *loggingT) output(s severity, log *logr.Logger, buf *buffer, depth int,
 			if l.file[s] == nil {
 				if err := l.createFiles(s); err != nil {
 					os.Stderr.Write(data) // Make sure the message appears somewhere.
+					l.mu.Unlock()         // At this point `logExitFunc` is nil (checked in #L1068), so `Unlock` needed.
```

Comment:
I don't understand that comment. #L1068 is the check in `loggingT.exit`, right?

Here we don't know whether `logExitFunc` is nil or not.

I can see how not unlocking can be problematic when `l.exit()` returns instead of killing the process. But under which circumstances can that happen? Can you explain which problem this change here is fixing?

Reply:
Yes, `logExitFunc` is checked  in `loggingT.exit`.
At this point we don't know whether `logExitFunc` is nil or not, so updated with setting it nil to unlock safely.
PTAL thanks.